### PR TITLE
#TP-13 feat: assign bus to routes

### PR DESCRIPTION
**Pull request by Delyc**
### What does this PR do?
Allow administrator and operator to assign a given bus to a given route
### Description of Task to be completed
[the tasks that you completed on the feature]
- [x] The operator is able to assign a bus to a route
- [x] The operator is able to see a list of bus to routes assignments
- [x] Appropriate errors are displayed
### How should this be manually tested?
[//]: # (Explain the why, not the how)
Follow the commands below
```
// Assuming you already cloned the project
git fetch origin
git checkout origin/ft-assign-buses-routes-TP-13
npm install
```

### CODE REVIEW

File: test/assign.test.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	const response = await chai
+		.request(app)
+		.post(`/api/v1/assign/bus-to-route/RAD000X/401`)
+		
+	
+		response.body.should.be.a('object');
+		response.body.should.have.property('status');
+		response.body.data.should.be.a('object');
+		response.body.message.should.be.a('string');
+		// response.body.message.should.be.eql("bus assigned successfully");
```
Comment (Jkadhuwa) [inline test/assign.test.js]: Why comment these assertions?

File: controllers/busRouteAssignController.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		return fail(res,404,null,'busNotExist',req) 
+	if (!route)
+		return fail(res,404,null,'routeNotExist',req) 
+
+		
+	bus.routeId = route.id;
+	bus.routecode = route.code;
+	await bus.save();
+		return success(res,200,bus,'busAssignedSuccessfully',req) 
+
```
Comment (Jkadhuwa) [inline controllers/busRouteAssignController.js:32]: - What happens when we assign a bus to another route when it's already assigned to a route? or
- How can I remove a bus from a specific route?

**Review by Jkadhuwa — DISMISSED** [review]
Good work @Delyc , left some comments below

File: controllers/busRouteAssignController.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { success, fail } from '../function/respond';
+import { buses as busModel, routes as routeModel } from '../models';
+
+
+export const Assign = async (req, res) => {
+	const { plate, code } = req.params;
```
Comment (chrisostome01) [inline controllers/busRouteAssignController.js]: I can see that we don't have any validation here and also this function may crush the server you may use try and catch that may fix it and validate all these inputs with Joi.

**Review by chrisostome01 — DISMISSED** [review]
_**well done @Delyc**_

File: routes/assign-bus-routes/assign.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ *        description: 
+ *    responses:
+ *      200:
+ *        description: successful
+ * 
+ */
+
+assignRouter.post(
+	'/bus-to-route/:plate/:code',
+	Assign
```
Comment (chanceDira) [inline routes/assign-bus-routes/assign.js]: It will be perfect if you protect this route.

**Review by chanceDira — COMMENTED** [review]
**_Good Work on this @Delyc._**

File: routes/assign-bus-routes/assign.js (hunk 4 above)
Comment (Delyc) [reply routes/assign-bus-routes/assign.js]: noted! hope now it is good

File: controllers/busRouteAssignController.js (hunk 2 above)
Comment (Delyc) [reply controllers/busRouteAssignController.js:32]: on the first point, when you tempt to assign a bus to another route it reassigns meaning it updates the current data by removing current route and assign to new assigned route while on the second point I have added a new route to handle that.

File: controllers/busRouteAssignController.js (hunk 3 above)
Comment (renepromesse) [reply controllers/busRouteAssignController.js]: @Delyc  did you work on this feedback?

Comment (Delyc) [reply controllers/busRouteAssignController.js]: no yet done, working on it rn

File: models/buses.js [hunk 5]
```diff
@@ -1,26 +1,32 @@
...
-  });
-  return buses;
-};
\ No newline at end of file
+	class buses extends Model {
+		/**
+		 * Helper method for defining associations.
+		 * This method is not a part of Sequelize lifecycle.
+		 * The `models/index` file will call this method automatically.
+		 */
```
Comment (renepromesse) [inline models/buses.js]: remove this commentd lines

File: test/route.test.js [hunk 6]
```diff
@@ -2,19 +2,19 @@ import chai from 'chai'
...
-		db.route.destroy({
-			where: {},
-			truncate: true,
-			restartIdentity: true,
-		});
+		// db.route.destroy({
+		// 	where: {},
+		// 	truncate: true,
+		// 	restartIdentity: true,
+		// });
```
Comment (renepromesse) [inline test/route.test.js]: remove this comments

**Review by renepromesse — CHANGES_REQUESTED** [review]
Good work @Delyc , I left some reviews,
- squash the two commits into 1
- add screenshots of the requests you're sending and the response

**Review by jeandamascyifuzo — APPROVED** [review]

**Review by renepromesse — APPROVED** [review]

**Pull request merged**